lib/spack/spack/fetch_strategy.py

```diff
@@ -1298,8 +1300,10 @@ def for_package_version(pkg, version):
     if version not in pkg.versions:
         return _extrapolate(pkg, version)
 
+    # Set package args first so version args can override them
+    args = {'fetch_options': pkg.fetch_options}
     # Grab a dict of args out of the package version dict
-    args = pkg.versions[version]
+    args.update(pkg.versions[version])
```

Comment:
I think what you say in https://github.com/spack/spack/pull/15317#issuecomment-597483399 works because of this update after initializing with `pkg.fetch_options`. I think a test should be added for this override behavior so it doesn't disappear in a future regression.

Also: can the `jdk` package use this to set `cookie`?

Side note: If a package version defines `fetch_options`, that dictionary will completely override the one provided by `Package.fetch_options`. I think that's for the best (since merging might be convenient but would make it hard to remove settings for some versions).

Reply:
I have pushed two new commits that should address your comments.